features/manage-position/Create.tsx

```diff
@@ -238,7 +245,14 @@ const Create = () => {
       {hash && (
         <Box py={2}>
           <Typography>
-            <strong>Tx Hash: </strong> {hash}
+            <strong>Tx Receipt: </strong>
+              <Link
+                href={getEtherscanUrl(hash)}
```

Comment:
The underlying implementation of `getEtherscanUrl` uses the `Connection` global state container, which is actually a `useContext` hook. Since hooks are not supposed to be called within conditionals, this may cause bugs in the future (if `hash` is null, `getEtherscanUrl` is never called).

I propose that you:

1. Convert `getEtherscanUrl.ts` into a custom hook `useEtherscanUrl.ts`, and;

2. Use it within the component before the returning of JSX.

For example:

```ts
// useEtherscanUrl.ts

import Connection from "../containers/Connection"

const useEtherscanUrl = (hash: string) => {
  const { network } = Connection.useContainer()
  const etherscanUrl =  ... /* insert implementation here */

  return etherscanUrl
}
```

```tsx
// Create.tsx

import useEtherscanUrl from "./useEtherscanUrl"

const Create = () => {
  const etherscanUrl = useEtherscanUrl(hash)
  return (
    <div>{etherscanUrl}</div>
  )
}
```

You'll also have to handle the `null` case, which I agree can be annoying.

Reply:
Great suggestion, I implemented this as as suggested.